src/test_runner/failure_persistence/file.rs

```diff
@@ -144,159 +233,118 @@ where
     }
 }
 
-impl FailurePersistence for FileFailurePersistence {
-    fn load_persisted_failures(&self, source_file: Option<&'static str>) -> Vec<[u32; 4]> {
-        let p = self.resolve(
-                source_file
-                    .and_then(|s| absolutize_source_file(Path::new(s)))
-                    .as_ref()
-                    .map(|cow| &**cow));
-
-        let path: Option<&PathBuf> = match p {
-            Some(ref inner_path) => Some(inner_path),
-            None => None,
-        };
-        let result: io::Result<Vec<[u32; 4]>> = path.map_or_else(
-            || Ok(vec![]),
-            |path| {
-                // .ok() instead of .unwrap() so we don't propagate panics here
-                let _lock = PERSISTENCE_LOCK.read().ok();
+fn parse_seed_line(mut line: String, path: &Path, lineno: usize) -> Option<Seed> {
+    // Remove anything after and including '#':
+    if let Some(comment_start) = line.find('#') {
+        line.truncate(comment_start);
+    }
 
-                let mut ret = Vec::new();
+    // Split by whitespace and ignore empty lines:
+    let parts = line.trim().split(char::is_whitespace).collect::<Vec<_>>();
+    let len = parts.len();
+    if len == 0 { return None; }
+
+    // Decide which parser to use depending on format:
+    let first_ok = parts[0] == "xs";
+    let parser = match len {
+        5 if first_ok => {
+            eprintln!("proptest: using old format parsing [u32; 4]. \
+                       This format will be removed in version 0.9");
```

Comment:
What is the user's recourse here? I'd rather support the old format indefinitely.

Reply:
My understanding is that when proptest is run, then the old format will be read in, and then saved to the same file using the new format, so the migration should be automatic?

Reply:
> the old format will be read in, and then saved to the same file using the new format

The regression file is append-only.

> Are you saying that the old format should be supported indefinitely parallel with the new format

Yes, that.

Reply:
I suppose if we support the old format indefinitely, there's not really a benefit for the new one.

Reply:
> The regression file is append-only.

Ah I see; I read the code incorrectly before.

> I suppose if we support the old format indefinitely, there's not really a benefit for the new one.

I suppose not;

My hope was to drop the `byte-order` dependency in the not too distant future by not supporting the old format beyond 0.9 so that we don't accumulate technical debt.

Would it be feasible to offer a separate program that performs the migration automatically perhaps?
i.e: If you point it to a `failure-persistence.txt` file, it will convert the file to the new format.

Reply:
> My hope was to drop the `byte-order` dependency in the not too distant future

I wouldn't be opposed to just having the two conversions in private functions in this crate, though the `byteorder` crate also doesn't seem like a particularly heavy dependency.

Reply:
So... I think these two definitions of `convert_to_new_format` should be equivalent; `convert_to_new_format` is the definition using the exact same logic as in `byteorder`.

https://play.rust-lang.org/?gist=0347f1b9fbf316aae1a85e48c6d79e4d&version=stable&mode=debug